lib/services/android-plugin-build-service.ts

```diff
@@ -0,0 +1,365 @@
+		result.push(dependenciesScope);
+
+		return result;
+	}
+
+	private getScope(scopeName: string, content: string) {
+		const indexOfScopeName = content.indexOf(scopeName);
+		let result = "";
+		const OPENING_BRACKET = "{";
+		const CLOSING_BRACKET = "}";
+		let openBrackets = 0;
+		let foundFirstBracket = false;
+
+		let i = indexOfScopeName;
+		while (i < content.length) {
+			const currCharacter = content[i];
+			if (currCharacter === OPENING_BRACKET) {
+				if (openBrackets === 0) {
+					foundFirstBracket = true;
+				}
+
+				openBrackets++;
+			}
+
+			if (currCharacter === CLOSING_BRACKET) {
+				openBrackets--;
+			}
+
+			result += currCharacter;
+
+			if (openBrackets === 0 && foundFirstBracket) {
+				break;
+			}
+
+			i++;
+		}
+
+		return result;
+	}
+
+	/**
+	 * Returns whether the build has completed or not
+	 * @param {Object} options
+	 * @param {string} options.pluginName - The name of the plugin. E.g. 'nativescript-barcodescanner'
+	 * @param {string} options.platformsAndroidDirPath - The path to the 'plugin/src/platforms/android' directory.
+	 * @param {string} options.aarOutputDir - The path where the aar should be copied after a successful build.
+	 * @param {string} options.tempPluginDirPath - The path where the android plugin will be built.
+	 */
+	public async buildAar(options: IBuildOptions): Promise<boolean> {
+		this.validateOptions(options);
+
+		// IDEA: Accept as an additional parameter the Android Support Library version from CLI, pass it on as -PsupportVersion later to the build
+		// IDEA: apply app.gradle here in order to get any and all user-defined variables
+		// IDEA: apply the entire include.gradle here instead of copying over the repositories {} and dependencies {} scopes
+
+		const shortPluginName = this.getShortPluginName(options.pluginName);
+		const newPluginDir = path.join(options.tempPluginDirPath, shortPluginName);
+		const newPluginMainSrcDir = path.join(newPluginDir, "src", "main");
+		const defaultPackageName = "org.nativescript." + shortPluginName;
+
+		// find manifest file
+		//// prepare manifest file content
+		const manifestFilePath = this.getManifest(options.platformsAndroidDirPath);
+		let updatedManifestContent;
+		let shouldBuildAar = false;
+
+		// look for AndroidManifest.xml
+		if (manifestFilePath) {
+			shouldBuildAar = true;
+		}
+
+		// look for android resources
+		const androidSourceSetDirectories = this.getAndroidSourceDirectories(options.platformsAndroidDirPath);
+
+		if (androidSourceSetDirectories.length > 0) {
+			shouldBuildAar = true;
+		}
+
+		// if a manifest OR/AND resource files are present - write files, build plugin
+		if (shouldBuildAar) {
+			this.$fs.ensureDirectoryExists(newPluginMainSrcDir);
+
+			if (manifestFilePath) {
+				let androidManifestContent;
+				try {
+					androidManifestContent = this.$fs.readFile(manifestFilePath).toString();
+				} catch (err) {
+					throw Error(
```

Comment:
can you use the new keyword here - all over the codebase we are using `throw new Error` instead of `throw Error`
Same is valid for other places in the code where you are using the `throw Error` syntax